packages/components/src/Profile/Component.purs

```diff
@@ -379,6 +428,63 @@ editAddress self =
               Tracking.changeAddress self.props.profile.cusno "error: unexpected error when updating address"
     updateAddress _ = pure unit
 
+editEmail :: Self -> JSX
+editEmail self =
+  DOM.form
+    { className: "profile--edit-email"
+    , children:
+        [ InputField.inputField
+            { type_: InputField.Email
+            , name: "email"
+            , "placeholder": self.props.profile.email
+            , value: self.state.email
+            , onChange: \newEmail -> case newEmail of
+                Just n -> self.setState _ { email = Just n }
+                _ -> pure unit
+            , label: Just "E-postadress"
+            , validationError: inputFieldErrorMessage $ validateField Email self.state.email []
+            }
+        , submitButton
+        , DOM.div { className: "profile--submit-buttons", children: [ iconClose self EditEmail ] }
+        ]
+    , onSubmit: Events.handler preventDefault $ \_ -> submitNewEmail $ validateEmailForm self.state.email
+    }
+  where
+    submitButton = iconSubmit $ isValid (validateEmailForm self.state.email)
+
+    validateEmailForm :: Maybe String -> ValidatedForm EmailFormFields (Maybe String)
+    validateEmailForm form =
+      validateField Email form []
+
+    submitNewEmail :: ValidatedForm EmailFormFields (Maybe String) -> Effect Unit
+    submitNewEmail = unV
+      (\errors -> Console.error "Could not submit email.")
+      updateEmail
+
+    updateEmail :: Maybe String -> Effect Unit
+    updateEmail (Just email) = do
+      self.setState _ { editEmail = Loading mempty }
+      if email == self.props.profile.email then
+        self.setState _ { editEmail = AsyncWrapper.Error "E-postadressen är den samma som den gamla." }
+        else Aff.launchAff_ do
+          newUser <- User.updateUser self.props.profile.uuid $ User.UpdateEmail { email: email }
+          case newUser of
+            Right u -> liftEffect do
+              self.props.onUpdate u
+              self.setState _ { editEmail = Success Nothing }
+              Tracking.changeEmail self.props.profile.cusno "success"
+            Left UniqueViolation -> do
+              liftEffect do
+                self.setState _ { editEmail = AsyncWrapper.Error "Den här e-postadressen används för ett annat konto. Vänligen ta kontakt med kundservice om du har frågor." }
+              throwError $ error "Unique violation when updating email."
```

Comment:
Don't know if we need to throw here actually. I guess it's enough we just show the error message to the user. This would end up in Sentry though, if that was your reasoning.

Reply:
I removed the throw.